packages/generate/src/queries.ts

```diff
@@ -0,0 +1,201 @@
+// tslint:disable
+import {createClient, adapter} from "edgedb";
+import type {ConnectConfig} from "edgedb/dist/conUtils";
+import type {CommandOptions} from "./commandutil";
+import {generateQueryType} from "./codecToType";
+import type {QueryType} from "./codecToType";
+// import chokidar from "chokidar";
+// import globby from "globby";
+import type {Target} from "./generate";
+import {Cardinality} from "edgedb/dist/ifaces";
+
+export async function generateQueryFiles(params: {
+  root: string | null;
+  options: CommandOptions;
+  connectionConfig: ConnectConfig;
+}) {
+  if (params.options.file && params.options.watch) {
+    throw new Error(`Using --watch and --file mode simultaneously is not
+currently supported.`);
+  }
+
+  // console.log(`Detecting project root...`);
+
+  const noRoot = !params.root;
+  const root = params.root ?? adapter.process.cwd();
+  if (noRoot) {
+    console.warn(
+      `No \`edgedb.toml\` found, using process.cwd() as root directory:
+   ${params.root}
+`
+    );
+  } else {
+    console.log(`Detected project root via edgedb.toml:`);
+    console.log("   " + params.root);
+  }
+  const client = createClient({
+    ...params.connectionConfig,
+    concurrency: 5
+  });
+
+  // watch mode: start watcher
+
+  // file mode: introspect all queries and generate one file
+  // generate one query per file
+
+  const matches = await getMatches(root);
+
+  console.log(`Generating files for following queries:`);
+  for (const path of matches) {
+    const prettyPath = "./" + adapter.path.posix.relative(root, path);
+    console.log(`   ${prettyPath}`);
+    const query = await adapter.readFileUtf8(path);
+    const types = await generateQueryType(client, query);
+    const files = await generateFiles({
+      target: params.options.target!,
+      path,
+      types
+    });
+    for (const file of files) {
+      await adapter.fs.writeFile(
+        file.path,
+        `${file.imports}\n\n${file.contents}`
+      );
+    }
+  }
+
+  // if (!params.options.watch) {
+  //   const matches = await getMatches(root);
+  // } else {
+  //   const watcher = chokidar.watch("**/*.edgeql", {
+  //     cwd: root
+  //   });
+
+  //   // const method = params.options.watch ? watcher.on : watcher.once;
+  //   return watcher.once("all", async (evt, path, stats) => {
+  //     if (["add", "change"].includes(evt)) {
+  //       const modifier = {
+  //         add: "Detected:  ",
+  //         change: "Modified:   ",
+  //         unlink: "Deleted:   "
+  //       }[evt as string];
+  //       console.log(`${modifier}: ${path}`);
+
+  //       const prettyPath = adapter.path.posix.relative(root, path);
+  //       console.log(`Query ${prettyPath}`);
+  //     } else if (evt === "unlink") {
+  //     }
+  //   });
+  // }
+
+  // find all *.edgeql files
+  // for query in queries:
+  //   generate output file
+}
+
+async function getMatches(root: string) {
+  return adapter.walk(root, {
+    match: [/\.edgeql$/],
```

Comment:
Could this match a file without a name (`.edgeql`) and cause issues later down the line?